src/common.c

```diff
@@ -107,28 +108,22 @@ mkdir_recursive(const char *dir)
 }
 
 gboolean
-copy_file(const char *const sourcepath, const char *const targetpath)
+copy_file(const char *const sourcepath, const char *const targetpath, const gboolean overwrite_existing)
 {
-    int ch;
-    FILE *source = fopen(sourcepath, "rb");
-    if (source == NULL) {
-        return FALSE;
-    }
-
-    FILE *target = fopen(targetpath, "wb");
-    if (target == NULL) {
-        fclose(source);
-        return FALSE;
-    }
-
-    while((ch = fgetc(source)) != EOF) {
-        fputc(ch, target);
+    GFile *source = g_file_new_for_path(sourcepath);
```

Comment:
Due to [documentation](https://developer.gnome.org/gio/stable/GFile.html#g-file-new-for-path), `source` and `dest` must be freed with `g_object_unref()`.